cloud-gateway-service/src/test/java/org/zowe/apiml/cloudgatewayservice/acceptance/TokenSchemeTest.java

```diff
@@ -0,0 +1,336 @@
+        List<HttpCookie> cookies = Optional.ofNullable(httpExchange.getRequestHeaders().get("Cookie"))
+            .orElse(Collections.emptyList())
+            .stream()
+            .map(HttpCookie::parse)
+            .flatMap(Collection::stream)
+            .filter(c -> StringUtils.equalsIgnoreCase(cookieName, c.getName()))
+            .collect(Collectors.toList());
+        assertTrue(cookies.size() <= 1);
+        return cookies.isEmpty() ? null : cookies.get(0).getValue();
+    }
+
+    @Nested
+    @AcceptanceTest
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    class HighAvailability extends AcceptanceTestWithMockServices {
+
+        private MockService zaasZombie;
+        private MockService zaasError;
+        private MockService zaasOk;
+
+        private MockService service;
+
+        private String getServiceUrl() {
+            return basePath + "/service/api/v1/test";
+        }
+
+        @BeforeAll
+        void createAllZaasServices() throws IOException {
+            // on the beginning prepare all as zombie, each test will decide
+            zaasError = mockService("gateway").scope(MockService.Scope.CLASS)
+                .addEndpoint(getTokenEndpoint())
+                    .responseCode(500)
+                .and().build();
+            zaasZombie = mockService("gateway").scope(MockService.Scope.CLASS)
+                .addEndpoint(getTokenEndpoint())
+                    .bodyJson(new ZaasTokenResponse())
+                .and().build();
+            zaasOk = mockService("gateway").scope(MockService.Scope.CLASS)
+                .addEndpoint(getTokenEndpoint())
+                    .bodyJson(OK_RESPONSE)
+                    .assertion(he -> assertEquals("service", he.getRequestHeaders().getFirst("x-service-id")))
+                .and().build();
+
+            // south-bound service - alive for all tests
+            service = mockService("service").scope(MockService.Scope.CLASS)
+                .authenticationScheme(getAuthenticationScheme())
+                .addEndpoint("/service/test")
+                    .assertion(he -> assertEquals(JWT, getCookie(he, COOKIE_NAME)))
+                .and().start();
+        }
+
+        @Test
+        void givenNoInstanceOfZosmf_whenCallingAService_thenReturn500() {
+            zaasZombie.stop();
+            zaasError.stop();
+            zaasOk.stop();
+
+            given().when().get(getServiceUrl()).then().statusCode(500);
+            assertEquals(0, service.getCounter());
+        }
+
+        @Test
+        void givenInstanceOfZosmf_whenCallingAService_thenReturn200() throws IOException {
+            zaasZombie.stop();
+            zaasError.stop();
+            zaasOk.start();
+
+            given().when().get(getServiceUrl()).then().statusCode(200);
+            assertEquals(1, service.getCounter());
+        }
+
+        @Test
+        void givenZombieAndOkInstanceOfZosmf_whenCallingAService_preventZombieOne() throws IOException {
```

Comment:
would it be possibile to maybe add another nested class (i.e. group by the action `callingAsService`, to avoid such long test names?

Reply:
These are old tests that were just made abstract, renamed, and reused for more (zoweJwt) testing. I wanted to avoid unnecessary refactoring. @pj892031 Should I refactor your tests? :-)

Reply:
You can, I have no issue with that ;)

Reply:
I would like just to mention, that there are acceptance tests. It is complicated to split these tests because of creating web servers. It could make it slow and flaky..